File: Data Structures/hash/hashset.py
# HashSet
a = set()  # O(1) for initialization

# Add elements to HashSet
a.add(1)  # O(1) on average
a.add(2)  # O(1) on average
a.add(3)  # O(1) on average
a.add(4)  # O(1) on average

# Check if an element exists in HashSet
if 1 in a:  # O(1) on average
    print(True)

# Remove element
a.remove(4)  # O(1) on average

# Remove duplicate characters with HashSet
string = 'aaaaaaabbbbbbbbbbccccccccddddddd'
sets = set(string)  # O(n), where n is the length of the string
print(sets)

File: Data Structures/hash/hashset.cpp
#include <iostream>
#include <unordered_set>

int main() {
std::unordered_set<int> hashset;

//insert into hashset
hashset.insert(1);
hashset.insert(2);
hashset.insert(3);

//remove element from hashset
hashset.erase(2);

//check if element exist
if(hashset.find(1) != hashset.end()){
std::cout << "element found" << "\n";
}

//display set element
for(int element : hashset){
std::cout << element << "\n";
}
}

